lib/tasks_private/instance_types.rake

```diff
@@ -1,215 +1,191 @@
-
-  new_instance_types_file << output_instances.map { |i| format_instance(i) }.join("\n")
-  new_instance_types_file << <<EOT
-  }.freeze
-
-  # Types that are no longer advertised
-  DISCONTINUED_TYPES = {
-EOT
-  # current discontinued types
-  output_instances = []
-  current_discontinued = ManageIQ::Providers::Amazon::InstanceTypes::DISCONTINUED_TYPES.dup
-  current_discontinued.values.each do |instance|
-    instance[:disabled] = true
-    output_instances << instance
-  end
+    JSON_CONTENT_RE = /\bapplication\/json\b/i
 
-  # new discontinued from current_available
-  discontinued_count = current_available.count
-  current_available.values.each do |instance|
-    instance[:disabled] = true
-    output_instances << instance
-  end
+    def http_get(uri, headers = {})
+      LOGGER.info("Making request to #{uri}")
 
-  # new discontinued from current_deprecated
-  discontinued_count += current_deprecated.count
-  current_deprecated.values.each do |instance|
-    instance[:disabled] = true
-    output_instances << instance
-  end
+      response = Net::HTTP.start(uri.host, :use_ssl => true) do |http|
+        http.get(uri.request_uri, headers)
+      end
 
-  new_instance_types_file << output_instances.map { |i| format_instance(i) }.join("\n")
-  new_instance_types_file << <<EOT
-  }.freeze
+      body = response.body if response.code_type.body_permitted?
+      json = response['content-type'] =~ JSON_CONTENT_RE || uri.path.end_with?('json')
+      data = JSON.parse(body) if json && body.present?
 
-  def self.instance_types
-    additional = Hash(Settings.ems.ems_amazon.try!(:additional_instance_types)).stringify_keys
-    disabled = Array(Settings.ems.ems_amazon.try!(:disabled_instance_types))
-    AVAILABLE_TYPES.merge(DEPRECATED_TYPES).merge(DISCONTINUED_TYPES).merge(additional).except(*disabled)
-  end
+      unless response.kind_of?(Net::HTTPSuccess)
+        LOGGER.error(data['message']) if data&.key?('message')
+        LOGGER.error("Error getting data from #{uri}")
+        response.error!
+      end
 
-  def self.all
-    instance_types.values
-  end
+      data.presence || body.presence
+    end
 
-  def self.names
-    instance_types.keys
-  end
-end
-EOT
+    def get_gh_data(uri_path, query_data = {})
+      uri = URI::HTTPS.build(
+        :host  => GH_API_HOST,
+        :path  => "/repos/#{SDK_REPO_PATH}/#{uri_path}",
+        :query => (query_data.to_query unless query_data.empty?),
+      )
+      http_get(uri, GH_API_HEADERS)
+    end
 
-  File.open(ManageIQ::Providers::Amazon::Engine.root.join('instance_types.rb'), 'w') do |f|
-    f.puts(new_instance_types_file)
-  end
+    def get_gh_file(file_path, sha: nil)
+      if sha.blank?
+        commits = get_gh_data('commits', :path => file_path, :page => 1, :per_page => 1)
+        sha = commits[0]['sha']
+        LOGGER.debug("Latest commit SHA of '#{file_path}' file is #{sha}")
+      end
+
+      GH_CACHE.fetch("#{file_path}.#{sha}".tr('/', '.')) do
```

Comment:
Can you explain the reason for the cache?  I don't understand why it's necessary.

Reply:
It's primary usage is for `AwsProductsDataCollector`, because those files are too big, but permanent in time (new prices/positions - new price list).

Reply:
@AlexanderZagaynov Yeah, after I made this comment, I ran it locally and saw the size of those files...they are enormous.